### PR TITLE
Fix fluent-bit extraVolumes and extraVolumeMounts migration for v2

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

Fill in your description here.

---

###### Testing performed

- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/helm/sumologic/upgrade-2.0.0.sh [hunk 1]
```diff
@@ -657,12 +657,17 @@ function migrate_fluent_bit() {
...
+  fi
 
-  yq w -i "${TEMP_FILE}" 'fluent-bit.extraVolumes[+].hostPath.path' '/var/lib/fluent-bit'
-  yq w -i "${TEMP_FILE}" 'fluent-bit.extraVolumes[0].hostPath.type' 'DirectoryOrCreate'
-  yq w -i "${TEMP_FILE}" 'fluent-bit.extraVolumes[0].name' 'tail-db'
+  local EXTRA_VOLUME_MOUNTS
+  readonly EXTRA_VOLUME_MOUNTS="$(yq r "${TEMP_FILE}" -- 'fluent-bit.extraVolumeMounts')"
+  if [[ -n "${EXTRA_VOLUME_MOUNTS}" ]]; then
+    yq w -i "${TEMP_FILE}" 'fluent-bit.extraVolumeMounts' --from <(echo "${EXTRA_VOLUME_MOUNTS}")
+  fi
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/upgrade-2.0.0.sh:670]: If I understand correctly, you copy `fluent-bit.extraVolumeMounts` from `TEMP_FILE` as `fluent-bit.extraVolumeMounts` to `TEMP_FILE`

File: tests/upgrade_v2_script/static/fluent_bit.output.yaml [hunk 2]
```diff
@@ -11,6 +11,27 @@ fluent-bit:
       valueFrom:
         fieldRef:
           fieldPath: metadata.namespace
+  extraVolumes:
```
Comment (perk-sumo) [inline tests/upgrade_v2_script/static/fluent_bit.output.yaml:14]: Isnt't input and output the same? 🤔

File: deploy/helm/sumologic/upgrade-2.0.0.sh (hunk 1 above)
Comment (pmalek-sumo) [reply deploy/helm/sumologic/upgrade-2.0.0.sh]: Let's pretend this didn't happen. PTAL now

File: tests/upgrade_v2_script/static/fluent_bit.output.yaml (hunk 2 above)
Comment (pmalek-sumo) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:14]: Let's pretend this didn't happen. PTAL now

Comment (pmalek-sumo) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:14]: Actually to answer your question: it is. But that's ok right?

https://github.com/helm/charts/blob/7e45e678e39b88590fe877f159516f85f3fd3f38/stable/fluent-bit/values.yaml#L198-L207

vs 

https://github.com/fluent/helm-charts/blob/e6805d4509868b76717677c7f7eecc4834d49d5a/charts/fluent-bit/values.yaml#L111-L113

**Review by sumo-drosiek — APPROVED** [review]
yay :tada:

File: tests/upgrade_v2_script/static/fluent_bit.output.yaml (hunk 2 above)
Comment (perk-sumo) [reply tests/upgrade_v2_script/static/fluent_bit.output.yaml:14]: 👍

**Review by perk-sumo — APPROVED** [review]

**Pull request merged**